graylog2-server/src/main/java/org/graylog2/indexer/indices/ShardsInfo.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.indexer.indices;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.commons.lang3.EnumUtils;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Locale;
+import java.util.function.Function;
+
+public record ShardsInfo(String index, int shard, ShardType shardType, State state, long docs, String store, InetAddress ip, String node ) {
+
+    public static ShardsInfo create(JsonNode jsonNode) throws UnknownHostException {
+
+        String index = jsonNode.get("index").asText();
+        int shard =jsonNode.get("shard").asInt();
+
+        String ipString = getValueOrDefault(jsonNode, "ip", JsonNode::asText, null);
+        InetAddress ip = ipString != null ? InetAddress.getByName(ipString) : null;
```

Comment:
I wonder if that's worth the complexity of having to handle UnknownHostException.
Couldn't we just return the string and let the consumer deal with potentially converting it?